docs/docs/guide/project/manifests.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Manifests
+description: Learn about the manifest files that Tuist uses to define projects and workspaces and configure the generation process.
+---
+
+# Manifests
+
+Tuist defaults to Swift files as the primary way to define projects and workspaces and configure the generation process. These files are referred to as **manifest files** throughout the documentation. 
+
+The decision of using Swift was inspired by the [Swift Package Manager](https://www.swift.org/documentation/package-manager/), which also uses Swift files to define packages. Thanks to the usage of Swift, we can leverage the compiler to validate the correctness of the content and reuse code across different manifest files, and Xcode to provide a first-class editing experience thanks to the syntax highlighting, auto-completion, and validation.
+
+> [!NOTE] CACHING
+> Since manifest files are Swift files that need to be compiled, Tuist caches the compilation results to speed up the parsing process. Therefore, you'll notice that the first time you run Tuist, it might take a bit longer to generate the project. Subsequent runs will be faster.
+
+## Project.swift
+
+The [`Project.swift`](/reference/project-description/project) manifest declares an Xcode project. The project gets generated in the same directory where the manifest file is located with the name indicated in the `name` property.
+
+```swift
+// Project.swift
+let project = Project(
+    name: "App",
+    targets: [
+        // ....
+    ]
+)
+```
+
+
+> [!WARNING] ROOT VARIABLES
+> The only variable that should be at the root of the manifest is `let project = Project(...)`. If you need to reuse code across various parts of the manifest, you can use Swift functions.
```

Comment:
Why this? We have a lot of top-level variable and never had issues with it AFAIK

Reply:
I'm not sure. I'll have to investigate the next time we get a report.